src/OVAL/oval_generator.c

```diff
@@ -138,9 +156,30 @@ void oval_generator_set_timestamp(struct oval_generator *generator, const char *
 	generator->timestamp = oscap_strdup(timestamp);
 }
 
+void oval_generator_update_timestamp(struct oval_generator *generator)
+{
+	time_t et;
+	struct tm *lt;
+	char timestamp[] = "yyyy-mm-ddThh:mm:ss";
```

Comment:
I don't think so, because the timestamp variable is used in sprintf function three lines below.
"Const" means that the string is immutable, but snpritntf modifies it.
Manual page describes its prototype:
`int snprintf(char *str, size_t size, const char *format, ...);`
it requires `char *str`.
Moreover this is not my implementation, it is only refactored from `oval_generator_new()`.